README.md
# bee-pagination
[![npm version](https://img.shields.io/npm/v/bee-pagination.svg)](https://www.npmjs.com/package/bee-pagination)
[![Build Status](https://img.shields.io/travis/tinper-bee/bee-pagination/master.svg)](https://travis-ci.org/tinper-bee/bee-pagination)
[![Coverage Status](https://coveralls.io/repos/github/tinper-bee/bee-pagination/badge.svg?branch=master)](https://coveralls.io/github/tinper-bee/bee-pagination?branch=master)

用于内容过多，分页显示内容

Pagination组件内部集成了子组件PaginationButton.PaginationButton并不暴露给用户。

## 使用

使用单独的bee-Pagination包
#### 组件引入
先进行下载bee-Pagination包
```
npm install --save bee-pagination
```
组件调用
```js
import Pagination from 'bee-pagination';
React.render(<div>
   <Pagination
        prev
        next
        first
        last
        ellipsis
        boundaryLinks
        items={20}
        maxButtons={5}
        activePage={this.state.activePage}
        onSelect={this.handleSelect.bind(this)} />
</div>, document.getElementById('target'));
```
#### 样式引入
- 可以使用link引入dist目录下bee-pagination.css
```
<link rel="stylesheet" href="./node_modules/build/Pagination.css">
```
- 可以在js中import样式
```js
import "./node_modules/src/Pagination.scss"
//或是
import "./node_modules/build/Pagination.css"
```

### API

|参数|说明|类型|默认值|
|:---|:----|:---|:------|
|items|总页数|number|1|
|activePage|哪一页是激活状态|number|1|
|onSelect|切换页的方法|func |() => {}|
|maxButtons|显示最多页数按钮|number|0|
|prev|显示上一页按钮|bool|false|
|next|显示下一页按钮|bool|false|
|first|显示第一页按钮|bool|false|
|last|显示最后一页按钮|bool|last|
|ellipsis|显示省略按钮|bool|false|
|boundaryLinks|显示边界按钮|bool|false|
|gap|按钮之间有间隔|bool|false|
|noBorder|不显示按钮边框|bool|false|
|size|按钮大小oneOf:lg,sm,''|string|''|
|showJump|是否显示跳页选择|bool|false(为了兼容老版本)|
|onDataNumSelect|选择每页多少条的回调函数|func|() => {}|
|dataNumSelect|每页多少条的下拉选择Option内容|array|['5','10','15','20']|
|dataNum|下拉选择的设定值的index|num|1|
|total|一共多少条|num|1|
|disabled|pagination不可用，不可点击|bool|false|
|btnType|pagination确认按钮的样式属性集合，具体参考组件Button|Object|{shape:'border'}|

#### 开发调试

```sh
$ git clone https://github.com/tinper-bee/bee-pagination
$ cd bee-pagination
$ npm install
$ npm run dev
```


CHANGELOG.md
## [3.0.5](https://github.com/tinper-bee/bee-pagination/compare/v3.0.4...v3.0.5) (2021-03-23)



## [3.0.4](https://github.com/tinper-bee/bee-pagination/compare/v3.0.3...v3.0.4) (2020-10-30)


### Bug Fixes

* 跳转页在失去焦点时进行页面跳转,跳转后跳转页面消失 #JCJS-5171 ([e7eb806](https://github.com/tinper-bee/bee-pagination/commit/e7eb806969d2958d6fa139813449d33ac2894d18)), closes [#JCJS-5171](https://github.com/tinper-bee/bee-pagination/issues/JCJS-5171)



## [3.0.3](https://github.com/tinper-bee/bee-pagination/compare/v3.0.2...v3.0.3) (2020-09-09)


### Bug Fixes

* #JCJS-3945 maxButton设置为1以后，点击下一页，页码没变 ([baa39cf](https://github.com/tinper-bee/bee-pagination/commit/baa39cff51810b1e49eb40a152234224cb44b057)), closes [#JCJS-3945](https://github.com/tinper-bee/bee-pagination/issues/JCJS-3945)



## [3.0.2](https://github.com/tinper-bee/bee-pagination/compare/v3.0.1...v3.0.2) (2020-08-13)


### Features

* 支持定制font-size ([f202fdb](https://github.com/tinper-bee/bee-pagination/commit/f202fdb7cb271fc18129a087bf24bb4ae4187676))



## [3.0.1](https://github.com/tinper-bee/bee-pagination/compare/v3.0.0...v3.0.1) (2020-08-11)



# [3.0.0](https://github.com/tinper-bee/bee-pagination/compare/v2.0.18...v3.0.0) (2020-08-06)



## [2.0.18](https://github.com/tinper-bee/bee-pagination/compare/v2.0.17...v2.0.18) (2020-08-04)


### Bug Fixes

* 改变total参数，分页按钮不变 ([87ef965](https://github.com/tinper-bee/bee-pagination/commit/87ef9659314abdbcd24df900966be2444e239222))



## [2.0.17](https://github.com/tinper-bee/bee-pagination/compare/v2.0.16...v2.0.17) (2020-04-24)


### Bug Fixes

* 自动化 ([67523bb](https://github.com/tinper-bee/bee-pagination/commit/67523bb1f7db4ac1d1347ee733c5914246bfbab8))


### Features

* 删除package-lock ([3b18e2a](https://github.com/tinper-bee/bee-pagination/commit/3b18e2ada93713a94e3abd8a08aef65dd7465dd9))
* 英文修改 ([58f136d](https://github.com/tinper-bee/bee-pagination/commit/58f136df593de638d95fb81e617d6ac9679838a8))



<a name="2.0.16"></a>
## [2.0.16](https://github.com/tinper-bee/bee-pagination/compare/v2.0.15...v2.0.16) (2020-03-09)



<a name="2.0.15"></a>
## [2.0.15](https://github.com/tinper-bee/bee-pagination/compare/v2.0.14...v2.0.15) (2020-02-05)


### Bug Fixes

* 只有一页的情况下，仍然可以上下翻页 ([94a8cd8](https://github.com/tinper-bee/bee-pagination/commit/94a8cd8))



<a name="2.0.14"></a>
## [2.0.14](https://github.com/tinper-bee/bee-pagination/compare/v2.0.13...v2.0.14) (2019-11-21)



<a name="2.0.13"></a>
## [2.0.13](https://github.com/tinper-bee/bee-pagination/compare/v2.0.12...v2.0.13) (2019-11-14)



<a name="2.0.12"></a>
## [2.0.12](https://github.com/tinper-bee/bee-pagination/compare/v2.0.11...v2.0.12) (2019-10-18)



<a name="2.0.11"></a>
## [2.0.11](https://github.com/tinper-bee/bee-pagination/compare/v2.0.10...v2.0.11) (2019-10-11)



<a name="2.0.10"></a>
## [2.0.10](https://github.com/tinper-bee/bee-pagination/compare/v2.0.9...v2.0.10) (2019-09-17)



<a name="2.0.9"></a>
## [2.0.9](https://github.com/tinper-bee/bee-pagination/compare/v2.0.8...v2.0.9) (2019-08-28)



<a name="2.0.8"></a>
## [2.0.8](https://github.com/tinper-bee/bee-pagination/compare/v2.0.7...v2.0.8) (2019-05-24)



<a name="2.0.7"></a>
## [2.0.7](https://github.com/tinper-bee/bee-pagination/compare/v2.0.6...v2.0.7) (2019-04-23)



<a name="2.0.6"></a>
## [2.0.6](https://github.com/tinper-bee/bee-pagination/compare/v2.0.5...v2.0.6) (2019-04-09)



<a name="2.0.5"></a>
## [2.0.5](https://github.com/tinper-bee/bee-pagination/compare/v2.0.4...v2.0.5) (2019-03-26)



<a name="2.0.4"></a>
## [2.0.4](https://github.com/tinper-bee/bee-pagination/compare/v2.0.3...v2.0.4) (2019-03-22)



<a name="2.0.3"></a>
## [2.0.3](https://github.com/tinper-bee/bee-pagination/compare/v2.0.2...v2.0.3) (2019-03-07)



<a name="2.0.2"></a>
## [2.0.2](https://github.com/tinper-bee/bee-pagination/compare/v2.0.1...v2.0.2) (2019-03-06)



<a name="2.0.1"></a>
## [2.0.1](https://github.com/tinper-bee/bee-pagination/compare/v2.0.0...v2.0.1) (2019-03-01)



<a name="2.0.0"></a>
# [2.0.0](https://github.com/tinper-bee/bee-pagination/compare/v1.1.15...v2.0.0) (2019-01-23)



<a name="1.1.15"></a>
## [1.1.15](https://github.com/tinper-bee/bee-pagination/compare/v1.1.14...v1.1.15) (2019-01-22)



<a name="1.1.14"></a>
## [1.1.14](https://github.com/tinper-bee/bee-pagination/compare/v1.1.13...v1.1.14) (2018-12-10)



<a name="1.1.13"></a>
## [1.1.13](https://github.com/tinper-bee/bee-pagination/compare/v1.1.12...v1.1.13) (2018-12-06)



<a name="1.1.12"></a>
## [1.1.12](https://github.com/tinper-bee/bee-pagination/compare/v1.1.11...v1.1.12) (2018-12-03)



<a name="1.1.11"></a>
## [1.1.11](https://github.com/tinper-bee/bee-pagination/compare/v1.1.9...v1.1.11) (2018-11-29)



<a name="1.1.9"></a>
## [1.1.9](https://github.com/tinper-bee/bee-pagination/compare/v1.1.7...v1.1.9) (2018-11-05)



<a name="1.1.7"></a>
## [1.1.7](https://github.com/tinper-bee/bee-pagination/compare/v1.1.6...v1.1.7) (2018-11-01)



<a name="1.1.6"></a>
## [1.1.6](https://github.com/tinper-bee/bee-pagination/compare/v1.1.5...v1.1.6) (2018-10-25)



<a name="1.1.5"></a>
## [1.1.5](https://github.com/tinper-bee/bee-pagination/compare/v1.1.4...v1.1.5) (2018-08-14)



<a name="1.1.4"></a>
## [1.1.4](https://github.com/tinper-bee/bee-pagination/compare/v1.1.3...v1.1.4) (2018-07-12)



<a name="1.1.3"></a>
## [1.1.3](https://github.com/tinper-bee/bee-pagination/compare/v1.1.2...v1.1.3) (2018-06-14)



<a name="1.1.2"></a>
## [1.1.2](https://github.com/tinper-bee/bee-pagination/compare/v1.1.1...v1.1.2) (2018-06-13)



<a name="1.1.1"></a>
## [1.1.1](https://github.com/tinper-bee/bee-pagination/compare/v1.1.0...v1.1.1) (2018-06-11)



<a name="1.1.0"></a>
# [1.1.0](https://github.com/tinper-bee/bee-pagination/compare/24094db...v1.1.0) (2018-06-08)


### Bug Fixes

* scss样式bug ([24094db](https://github.com/tinper-bee/bee-pagination/commit/24094db))
* 修改当总共页数和最大按钮数只差一个时显示出错的bug ([d84efbe](https://github.com/tinper-bee/bee-pagination/commit/d84efbe))

src/Pagination.js
import classnames from "classnames";
import React from "react";
import PaginationButton from "./PaginationButton";
import Button from 'bee-button';
import Select from 'bee-select';
import Icon from 'bee-icon';
import assign from 'object-assign';
import cookie from 'react-cookies';
import PropTypes from "prop-types";
import i18n from './i18n';
import { getComponentLocale } from 'bee-locale/build/tool';
import omit from 'omit.js';


const Option = Select.Option;

const propTypes = {
    /**
     * 当前激活状态页
     */
    activePage: PropTypes.number,
    /**
     * 总页数
     */
    items: PropTypes.number,
    /**
     * 显示按钮从1到maxButton的按钮数
     */
    maxButtons: PropTypes.number,

    /**
     * 当为true,不管切换到多少页都显示第一页和最后一页的按钮
     */
    boundaryLinks: PropTypes.bool,

    /**
     * 当为true,显示省略号，否则
     *
     */
    ellipsis: PropTypes.oneOfType([PropTypes.bool, PropTypes.node]),

    /**
     *  当为true,显示点击到第一页的按钮
     */
    first: PropTypes.oneOfType([PropTypes.bool, PropTypes.node]),

    /**
     *  当为true,显示点击到最后一页的按钮
     */
    last: PropTypes.oneOfType([PropTypes.bool, PropTypes.node]),

    /**
     * 当为true,显示前一页按钮
     */
    prev: PropTypes.oneOfType([PropTypes.bool, PropTypes.node]),

    /**
     * 当为true,显示下一页按钮
     */
    next: PropTypes.oneOfType([PropTypes.bool, PropTypes.node]),

    /**
     * 暴露给用户的切换页的方法
     */
    onSelect: PropTypes.func,

    /**
     * You can use a custom element for the buttons
     */
    buttonComponentClass: PropTypes.oneOfType([
        PropTypes.element,
        PropTypes.string
    ]),
    /**
     * 每页多少条的选择
     */
    dataNumSelect: PropTypes.array,
    /**
     * 每页多少条选择哪一个
     */
    dataNum: PropTypes.number,
    /**
     * 显示跳页
     */
    showJump: PropTypes.bool,
    /**
     * 显示总共条数
     */
    total: PropTypes.number,
    /** 
     *  pagiantion不可点
     */
    disabled: PropTypes.bool,
    /**
     *  确认按钮的样式集合
     */
    btnType: PropTypes.object,
    /**
     *  渲染确认按钮的dom
     */
    confirmBtn: PropTypes.func,

    /**
     * 通过cookie来确定分页的size,需设定唯一的key值
     */
    sizeWithCookie: PropTypes.string,
};


const defaultProps = {
    activePage: 1,
    maxButtons: 0,
    first: false,
    last: false,
    prev: false,
    next: false,
    ellipsis: true,
    boundaryLinks: false,
    clsPrefix: "u-pagination",
    gap: false,
    noBorder: false,
    dataNumSelect: [
        '5',
        '10',
        '15',
        '20'
    ],
    // dataNum: 1,
    showJump: false,
    locale: {},
    disabled: false,
    btnType: { shape: 'border' },
    confirmBtn: () => {},
    sizeWithCookie: '',
};


class Pagination extends React.Component {
    constructor(props, context) {
        super(props, context);
        let size = props.dataNumSelect.findIndex(item=>String(item) === String(cookie.load(props.sizeWithCookie)));
        let dataNum = size === -1 ? (props.dataNum!=undefined?props.dataNum:1) : size;
        this.state = {
            activePage: this.props.activePage,//当前的页码
            dataNum: dataNum,
            items: props.items ? props.items : props.total ? Math.ceil(props.total / props.dataNumSelect[dataNum]) : 1,
            jumpPageState: ''
        }
    }

    componentWillReceiveProps(nextProps) {
        let { activePage, dataNum, sizeWithCookie, dataNumSelect, total,items} = nextProps;
        if (this.state.activePage !== activePage) {
            this.setState({
                activePage: activePage,
            })
        }
        // 判断条件从 this.props.dataNum !== dataNum  改为 this.state.dataNum !== dataNum; Hua
        if ((dataNum === 0 || dataNum) && this.state.dataNum !== dataNum) {
            if(sizeWithCookie){
                cookie.save(sizeWithCookie,dataNumSelect[dataNum],{maxAge: 60*60*24*7});
            }
            this.setState({
                // dataNum: dataNum,
                // 20181210因为dataNumSelect的某项不是数字或者数字字符串
                items: Number.isNaN(parseInt(dataNumSelect[dataNum])) ? 1:Math.ceil(total / parseInt(dataNumSelect[dataNum]))
            })
        }
        if ('items' in nextProps && this.props.items !== items) {
            let newItems = items === 0 ? 1 : items;
            this.setState({
                items: newItems,
            })
        }
        if ('total' in nextProps && this.props.total !== total) {
            let pageSize = parseInt(dataNumSelect[dataNum]);
            let defaultPageSize = parseInt(dataNumSelect[1]);
            this.setState({
                items: Number.isNaN(pageSize) ? Math.ceil(total / defaultPageSize):Math.ceil(total / pageSize)
            })
        }
    }

    onKeyup = (e) => {
        e.keyCode === 13 && this.handleEnsurePageJump()
    }

    setPageJump = (e) => {
        let value = e.target.value;
        // 20181129跳转内容可以清空
        if (value !== '' && (isNaN(Number(value)) || value > this.state.items || value <= 0)) {
            return false;
        } else {
            this.setState({
                jumpPageState: value
            })
        }

    }

    /**
     * 确认跳页
     */
    handleEnsurePageJump = () => {
        const { jumpPageState } = this.state;
        const { onSelect } = this.props;
        if (jumpPageState === '') {
            return;
        }
        this.setState({
            activePage: jumpPageState * 1,
            jumpPageState: ''
        });
        if (typeof onSelect === 'function') {
            onSelect(jumpPageState * 1)
        }
    }

    /**
     * 每页多少条的选择
     * @param e
     */
    dataNumSelect = (value) => {
        // console.log(value);
        const { onDataNumSelect, total, sizeWithCookie } = this.props;
        let dataNumValue = this.props.dataNumSelect[value];
        // console.log("dataNumValue", dataNumValue);
        if (total) {
            this.setState({
                // 20181210因为dataNumSelect的某项不是数字或者数字字符串
                items: Number.isNaN(parseInt(dataNumValue))?1:Math.ceil(total / dataNumValue)
            })
        }
        this.setState({
            dataNum: value
        })
        // 塞cookie
        if(sizeWithCookie){
            cookie.save(sizeWithCookie,dataNumValue,{maxAge: 60*60*24*7});
        }
        if (typeof onDataNumSelect === 'function') {
            onDataNumSelect(value, dataNumValue)
        }
    }

    /**
     * 渲染page的按钮
     * @param activePage
     * @param items
     * @param maxButtons
     * @param boundaryLinks
     * @param ellipsis
     * @param buttonProps
     * @returns {Array}
     */
    renderPageButtons(activePage, items, maxButtons, boundaryLinks, ellipsis, buttonProps) {
        let pageButtons = [];

        let startPage;
        let endPage;
        let hasHiddenPagesAfter;

        if (maxButtons) {
            //根据max很当前activepage计算出应隐藏activeButton之前的页数
            let hiddenPagesBefore = activePage - parseInt(maxButtons / 2, 10);
            startPage = hiddenPagesBefore > 2 ? hiddenPagesBefore : 1;
            //计算出是否存在隐藏activeButton之后的页数
            hasHiddenPagesAfter = startPage + maxButtons < items + 1;

            if (!hasHiddenPagesAfter) {
                endPage = items;
                startPage = items - maxButtons + 1;
                if (startPage < 1) {
                    startPage = 1;
                }
            } else {
                endPage = startPage + maxButtons - 1;
            }
        } else {
            startPage = 1;
            endPage = items;
        }
        //将所有的button循环渲染出来
        for (let pagenumber = startPage; pagenumber <= endPage; pagenumber++) {
            pageButtons.push(
                <PaginationButton
                    {...buttonProps}
                    key={pagenumber}
                    eventKey={pagenumber}
                    active={pagenumber === activePage}
                >
                    {pagenumber}
                </PaginationButton>
            );
        }
        //如果boundaryLinks和eclipsis且startPage!=1 需要加上before More Button
        if (boundaryLinks && ellipsis && startPage !== 1) {
            if (startPage > 2) {
                pageButtons.unshift(
                    <PaginationButton
                        key="ellipsisFirst"
                        disabled
                        componentClass={buttonProps.componentClass}
                        className={`${this.props.clsPrefix}-item-ellipsis`}
                    >
                        <span aria-label="More">
                            {ellipsis === true ? "\u2026" : ellipsis}
                        </span>
                    </PaginationButton>
                );
            }
            //加上最小边界 Button
            pageButtons.unshift(
                <PaginationButton {...buttonProps} key={1} eventKey={1} active={false}>
                    1
                </PaginationButton>
            );
        }
        //如果maxButtons和eclipsis且hasHiddenPagesAfter 需加上after More Button
        if (maxButtons && hasHiddenPagesAfter && ellipsis) {
            if (!boundaryLinks || (boundaryLinks && (items > 1 + endPage))) {
                pageButtons.push(
                    <PaginationButton
                        key="ellipsis"
                        disabled
                        componentClass={buttonProps.componentClass}
                        className={`${this.props.clsPrefix}-item-ellipsis`}
                    >
                        <span aria-label="More">
                            {ellipsis === true ? "\u2026" : ellipsis}
                        </span>
                    </PaginationButton>
                );
            }
            //如果最后一个页数按钮不等于总页数 且 边界为true 需加上最大边界按钮
            if (boundaryLinks && endPage !== items) {
                pageButtons.push(
                    <PaginationButton
                        {...buttonProps}
                        key={items}
                        eventKey={items}
                        active={false}
                    >
                        {items}
                    </PaginationButton>
                );
            }
        }

        return pageButtons;
    }

    render() {
        const local = getComponentLocale(this.props, this.context, 'Pagination', () => i18n);
        const {
            items,
            maxButtons,
            boundaryLinks,
            ellipsis,
            first,
            last,
            prev,
            next,
            onSelect,
            buttonComponentClass,
            noBorder,
            className,
            clsPrefix,
            size,
            gap,
            onDataNumSelect,
            dataNumSelect,
            // dataNum,
            activePage,
            showJump,
            total,
            disabled,
            btnType,
            confirmBtn,
            ...others
        } = this.props;

        const activePageState = this.state.activePage;
        const jumpPageState = this.state.jumpPageState;

        const btnDom = confirmBtn(assign({}, this.props));
 
        const classes = {};
        if (noBorder) {
            classes[`${clsPrefix}-no-border`] = true;
        }
        if (size) {
            classes[`${clsPrefix}-${size}`] = true;
        }
        if (gap) {
            classes[`${clsPrefix}-gap`] = true;
        }
        if (!!btnDom) {
            classes[`${clsPrefix}-with-jumpbtn`] = true;
        }


        let classNames = classnames(`${clsPrefix}-list`, classes);
        const wrapperClass = classnames(clsPrefix, {
            [`${clsPrefix}-disabled`]: disabled,
        })
        /**
         *  页按钮属性
         *  onSelect:暴露在外层交互动作，也是与父组件Pagination的交流接口
         *  componentClass: 用户定义的按钮dom元素类型
         */
        const buttonProps = {
            onSelect,
            componentClass: buttonComponentClass
        };
        const jumpBtn = btnDom ? (
            <div className="page_jump_btn" onClick={this.handleEnsurePageJump}>
              {btnDom}
            </div>
        ) : null;
        return (

            <div className={classnames(wrapperClass,className)}>
                {
                    disabled && <div className={`${clsPrefix}-disabled-mask`}>
                    </div>
                }
                <ul {...omit(others, ['dataNum','sizeWithCookie'])} className={classNames}>
                    {first && (
                        <PaginationButton
                            {...buttonProps}
                            eventKey={1}
                            disabled={activePageState === 1}
                            iconBtn={true}
                            className={`${clsPrefix}-first`}
                        >
                            <Icon type="uf-2arrow-left"></Icon>
                            {/* <span aria-label="First">{first === true ? "\u00ab" : first}</span> */}
                        </PaginationButton>
                    )}
                    {prev && (
                        <PaginationButton
                            {...buttonProps}
                            eventKey={activePageState - 1}
                            disabled={activePageState === 1}
                            iconBtn={true}
                            className={`${clsPrefix}-prev`}
                        >
                            <Icon type="uf-arrow-left"></Icon>
                            {/* <span aria-label="Previous">{prev === true ? "\u2039" : prev}</span> */}
                        </PaginationButton>
                    )}

                    {this.renderPageButtons(
                        activePageState,
                        this.state.items,
                        maxButtons,
                        boundaryLinks,
                        ellipsis,
                        buttonProps
                    )}

                    {next && (
                        <PaginationButton
                            {...buttonProps}
                            eventKey={activePageState + 1}
                            disabled={activePageState >= this.state.items}
                            iconBtn={true}
                            className={`${clsPrefix}-next`}
                            >
                            <Icon type="uf-arrow-right"></Icon>
                            {/* <span aria-label="Next">{next === true ? "\u203a" : next}</span> */}
                        </PaginationButton>
                    )}
                    {last && (
                        <PaginationButton
                            {...buttonProps}
                            eventKey={this.state.items}
                            disabled={activePageState >= this.state.items}
                            iconBtn={true}
                            className={`${clsPrefix}-last`}
                            >
                            <Icon type="uf-2arrow-right"></Icon>
                            {/* <span aria-label="Last">{last === true ? "\u00bb" : last}</span> */}
                        </PaginationButton>
                    )}
                </ul>
                {
                    total != null ? (
                        <div className={`${clsPrefix}-total`}>
                            {local['total']}
                            <span>{total}</span>
                            {local['items']}
                        </div>
                    ) : null
                }

                {
                    showJump ? (
                        <div className="data_per_select">
                            <span>{local['show']}</span>
                            {/* <select
                                    name="data-select"
                                    className="data_select"
                                    value={this.state.dataNum}
                                    onChange={this.dataNumSelect}>
                                    {dataNumSelect.length > 0 &&
                                    dataNumSelect.map((item, i) => {
                                        return <option key={i} value={i}>{item}</option>
                                    })}
                                </select> */}
                            <Select
                                // className="data_select"
                                dropdownClassName={`${clsPrefix}-data_per_select`}
                                value={this.state.dataNum}
                                onChange={this.dataNumSelect}>
                                {dataNumSelect.length > 0 &&
                                    dataNumSelect.map((item, i) => {
                                        return <Option key={i} value={i}>{item}</Option>
                                    })}
                            </Select>
                            <span>{local['items']}</span>
                        </div>
                    ) : null
                }
                {
                    showJump ? (
                        <div className="page_jump">
                            <span>{local['goto']}</span>
                            <input
                                className="page_jump_value"
                                value={jumpPageState}
                                onKeyDown={this.onKeyup}
                                onChange={this.setPageJump}
                            />
                            <span>{local['page']}</span>
                            {jumpBtn}
                            {/* <Button
                                className="page_jump_btn"
                                onClick={this.handleEnsurePageJump}
                                // shape="border"
                                {...btnType}
                            >
                            <span>{local['ok']}</span>
                            </Button> */}
                        </div>
                    ) : null
                }

            </div>

        );
    }
}

Pagination.propTypes = propTypes;
Pagination.defaultProps = defaultProps;
Pagination.contextTypes = {
    beeLocale: PropTypes.object
};
export default Pagination;